gui/src/app/pyodide/AnalysisPyFileEditor.tsx

```diff
@@ -0,0 +1,169 @@
+        divElement.style.color = "blue";
+        const preElement = document.createElement("pre");
+        divElement.appendChild(preElement);
+        preElement.textContent = x;
+        consoleOutputDiv?.appendChild(divElement);
+      },
+      onStderr: (x) => {
+        console.error(x);
+        const divElement = document.createElement("div");
+        divElement.style.color = "red";
+        const preElement = document.createElement("pre");
+        divElement.appendChild(preElement);
+        preElement.textContent = x;
+        consoleOutputDiv?.appendChild(divElement);
+      },
+      onStatus: (status) => {
+        setStatus(status);
+      },
+      onImage: (image) => {
+        const b64 = image;
+        const imageUrl = `data:image/png;base64,${b64}`;
+
+        const img = document.createElement("img");
+        img.src = imageUrl;
+
+        const divElement = document.createElement("div");
+        divElement.appendChild(img);
+        imageOutputDiv?.appendChild(divElement);
+      },
+    });
+    setAnalysisPyWorker(worker);
+    return () => {
+      worker.destroy();
+    };
+  }, [consoleOutputDiv, imageOutputDiv]);
+
+  const handleRun = useCallback(async () => {
+    if (status === "running") {
+      return;
+    }
+    if (editedFileContent !== fileContent) {
+      throw new Error("Cannot run edited code");
+    }
+    if (!analysisPyWorker) {
+      throw new Error("analysisPyWorker is not defined");
+    }
+    if (consoleOutputDiv) {
+      consoleOutputDiv.innerHTML = "";
+    }
+    if (imageOutputDiv) {
+      imageOutputDiv.innerHTML = "";
+    }
+    analysisPyWorker.run(fileContent);
+  }, [editedFileContent, fileContent, status, analysisPyWorker, consoleOutputDiv, imageOutputDiv]);
+  const toolbarItems: ToolbarItem[] = useMemo(() => {
+    const ret: ToolbarItem[] = [];
+    const runnable = fileContent === editedFileContent && imageOutputDiv;
+    if (runnable) {
+      ret.push({
+        type: "button",
+        tooltip: "Run script",
+        label: "Run",
+        icon: <PlayArrow />,
+        onClick: handleRun,
+        color: "black",
+      });
+    }
+    if (!imageOutputDiv) {
+      ret.push({
+        type: "text",
+        label: "No output window",
+        color: "red",
+      });
+    }
+    let label: string;
+    let color: string;
+    if (status === "loading") {
+      label = "Loading pyodide...";
+      color = "blue";
+    } else if (status === "running") {
+      label = "Running...";
+      color = "blue";
+    } else if (status === "completed") {
+      label = "Completed";
+      color = "green";
+    } else if (status === "failed") {
+      label = "Failed";
+      color = "red";
+    } else {
+      label = "";
+      color = "black";
+    }
+
+    if (label) {
+      ret.push({
+        type: "text",
+        label,
+        color,
+      });
+    }
+    return ret;
+  }, [fileContent, editedFileContent, handleRun, status, imageOutputDiv]);
```

Comment:
I agree, but I think we should hold off until things solidify more, since we don't yet know exactly what will be in common (or different) between the 4 cases.